stats/opentelemetry/csm/observability.go

```diff
@@ -0,0 +1,95 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package csm
+
+import (
+	"context"
+	"net/url"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/internal"
+	"google.golang.org/grpc/stats/opentelemetry"
+	otelinternal "google.golang.org/grpc/stats/opentelemetry/internal"
+)
+
+var clientSideOTelWithCSM grpc.DialOption
+var clientSideOTel grpc.DialOption
+
+// Observability sets up CSM Observability for the binary globally. It sets up
+// two client side OpenTelemetry instrumentation components configured with the
+// provided options, one with a CSM Plugin Option configured, one without.
+// Created channels will pick up one of these dependent on whether channels are
+// CSM Channels or not, which is derived from the Channel's parsed target after
+// dialing.
+//
+// It sets up a server side OpenTelemetry instrumentation component configured
+// with options provided alongside a CSM Plugin Option to be registered globally
+// and picked up for every server.
+//
+// The CSM Plugin Option is instantiated with local labels and metadata exchange
+// labels pulled from the environment, and emits metadata exchange labels from
+// the peer and local labels. Context timeouts do not trigger an error, but set
+// certain labels to "unknown".
+//
+// This function is not thread safe, and should only be invoked once in main
+// before any channels or servers are created. Returns a cleanup function to be
+// deferred in main.
```

Comment:
What's the reason to return a cleanup function if it needs to be called in main?  Can we just remove it?

Reply:
Doesn't need to clean up resources? It will leave globals around but I guess this is fine if your binary is exiting? I returned this to be like gcp/observability.

Reply:
Seems like we're keeping this since we might be doing other kinds of exporting in the future that will need to be flushed for correctness.